internal/pkg/describe/backend_app.go

```diff
@@ -0,0 +1,190 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package describe
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"text/tabwriter"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy/cloudformation/stack"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+)
+
+// BackendAppDescriber retrieves information about a backend application.
+type BackendAppDescriber struct {
+	app             *archer.Application
+	enableResources bool
+
+	store            storeSvc
+	appDescriber     appDescriber
+	initAppDescriber func(string) error
+}
+
+// NewBackendAppDescriber instantiates a backend application describer.
+func NewBackendAppDescriber(project, app string) (*BackendAppDescriber, error) {
+	svc, err := store.New()
+	if err != nil {
+		return nil, fmt.Errorf("connect to store: %w", err)
+	}
+	meta, err := svc.GetApplication(project, app)
+	if err != nil {
+		return nil, err
+	}
+	opts := &BackendAppDescriber{
+		app:             meta,
+		enableResources: false,
+		store:           svc,
+	}
+	opts.initAppDescriber = func(env string) error {
+		d, err := NewAppDescriber(project, env, app)
+		if err != nil {
+			return err
+		}
+		opts.appDescriber = d
+		return nil
+	}
+	return opts, nil
+}
+
+// NewBackendAppDescriberWithResources instantiates a backend application with stack resources.
+func NewBackendAppDescriberWithResources(project, app string) (*BackendAppDescriber, error) {
+	d, err := NewBackendAppDescriber(project, app)
+	if err != nil {
+		return nil, err
+	}
+	d.enableResources = true
+	return d, nil
+}
+
+// URI is used to make BackendAppDescriber have the same signature as WebAppDescriber
+func (d *BackendAppDescriber) URI(envName string) (string, error) {
+	s := serviceDiscovery{}
+	return s.String(), nil
```

Comment:
Shouldn't this be like 😂:
```go
		if err := d.initAppDescriber(env.Name); err != nil {
			return nil, err
		}
		appParams, err := d.appDescriber.Params()
		if err != nil && !IsStackNotExistsErr(err) {
			return nil, fmt.Errorf("retrieve application deployment configuration: %w", err)
		}
		if err != nil {
			continue
		}
		return serviceDiscovery{
			AppName:     d.app.Name,
			Port:        appParams[stack.LBWebAppContainerPortParamKey],
			ProjectName: d.app.Project,
		}.String()
```

Reply:
lol I thought this to be a mock function